src/agent/src/main.cpp

```diff
@@ -1,70 +1,51 @@
-#include <agent.hpp>
-#include <agent_info.hpp>
 #include <cmd_ln_parser.hpp>
-#include <http_client.hpp>
 #include <logger.hpp>
-#include <register.hpp>
-
-#include <optional>
+#include <process_options.hpp>
 
 int main(int argc, char* argv[])
 {
     Logger logger;
-    LogInfo("Starting Wazuh Agent.");
+    CommandlineParser cmdParser(argc, argv);
+
+    LogInfo("Starting Wazuh Agent service.");
 
     try
     {
-        CommandlineParser cmdParser(argc, argv);
-
-        if (cmdParser.OptionExists("--register"))
+        if (cmdParser.OptionExists("register"))
         {
-            LogInfo("Starting registration process");
-
-            if (cmdParser.OptionExists("--user") && cmdParser.OptionExists("--password") &&
-                cmdParser.OptionExists("--key"))
-            {
-                const auto user = cmdParser.GetOptionValue("--user");
-                const auto password = cmdParser.GetOptionValue("--password");
-
-                AgentInfo agentInfo;
-                agentInfo.SetKey(cmdParser.GetOptionValue("--key"));
-
-                if (cmdParser.OptionExists("--name"))
-                {
-                    agentInfo.SetName(cmdParser.GetOptionValue("--name"));
-                }
-                else
-                {
-                    agentInfo.SetName(boost::asio::ip::host_name());
-                }
-
-                http_client::HttpClient httpClient;
-                const registration::UserCredentials userCredentials {user, password};
-
-                if (registration::RegisterAgent(userCredentials, httpClient))
-                {
-                    LogInfo("Agent registered.");
-                }
-                else
-                {
-                    LogError("Registration fail.");
-                }
-            }
-            else
-            {
-                LogError("--user, --password and --key args are mandatory");
-            }
-
-            LogInfo("Exiting ...");
-            return 0;
+            RegisterAgent(cmdParser.GetOptionValue("--user"),
+                          cmdParser.GetOptionValue("--password"),
+                          cmdParser.GetOptionValue("--key"),
+                          cmdParser.GetOptionValue("--name"));
+        }
+        else if (cmdParser.OptionExists("start"))
+        {
+            StartAgent();
         }
+        else if (cmdParser.OptionExists("restart"))
+        {
+            StopAgent();
+            StartAgent();
+            LogInfo("Wazuh Agent restarted");
```

Comment:
Maybe this log is not necessary (stop and star already write logs).

Reply:
Done